kernel_gateway/notebook_http/__init__.py

```diff
@@ -122,7 +122,8 @@ def create_request_handlers(self):
             handler_args = { 'sources' : verb_source_map,
                 'response_sources' : response_source_map,
                 'kernel_pool' : self.kernel_pool,
-                'kernel_name' : self.parent.kernel_manager.seed_kernelspec
+                'kernel_name' : self.parent.kernel_manager.seed_kernelspec,
+                'kernel_language' : getattr(self.parent.kernel_spec_manager.get_kernel_spec(self.parent.kernel_manager.seed_kernelspec), 'language', 'no_spec')
```

Comment:
I'm not sure that this is a good place for the lookup. Can't the handler do the lookup itself?
If there is a need to set kernel_language from outside in some cases, then give it a default of null or the empty string, and let the handler deal with that. If there is no need to set kernel_language from outside, then don't add it to the handler_args at all.

Reply:
I think it as same as other args in handler_args, that's why we construct a `handler_args` here, we just pass what we need and not a some object and then get just a needed attribute from it. 

And I read through the code once more and find that we use the `lang` extract from notebook's metadata to determine the comment sign:
https://github.com/jupyter/kernel_gateway/blob/05e714805e5e74e1b4dbcbe463e4ab8ac295c490/kernel_gateway/notebook_http/__init__.py#L65-L69

So I think maybe I should also use this same `lang` to keep it consistent.  And here we can 
 make `lang` as an instance variable.

Reply:
The difference to the other args is that `kernel_language` is dependent on `kernel_name`.

Reply:
And I ever thought that `kernel_language` is dependent on `kernel_name`, but now I don't think so. Since I found that we can get the `language` directly like the above mentioned:

https://github.com/jupyter/kernel_gateway/blob/05e714805e5e74e1b4dbcbe463e4ab8ac295c490/kernel_gateway/notebook_http/__init__.py#L65-L69

And I think we should keep it consistent, i.e. get the `language` from the same source, the notebook's metadata.

Reply:
Oh, I see. That's one of the cases where I was missing the big picture :-)
Yes, the common source of information is the notebook, not the kernel_name.
Then it makes sense to pass both values as separate arguments.